hooks/__tests__/useDetectOS.test.ts

```diff
@@ -0,0 +1,35 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { useDetectOS } from '../useDetectOS';
+
+const mockNavigator = {
```

Comment:
Let's try multiple userAgent's and userAgentData's. I bet there's a list somewhere out on the internet.

You would keep iterating these user agent data's and iterating the tests on demand.

**Pretty much:**

- You move the mocking of the navigator to inside the `it`
- Pretty much foreach the array/collection
- Keep asserting that expect gives the appropriate value

Reply:
Ah, I just saw you tested this on `detectOS.ts`; no need to do it here then.